### PR TITLE
asserts,overlord/devicestate: simplify private key/key pairs APIs, they take just key ids

### PR DESCRIPTION
**Pull request by pedronis**
As discussed in the context of the device registration code, this simplifies the KeypairManager API and the key pair methods of Database to take only key ids and not authority ids, given that key ids are now really meant to be globally unique.

### CODE REVIEW

**Comment by mvo5** [comment]
Code looks fine to me, I like how it makes the code simpler but I do not know enough about the original motivation for the previous API to be fully confident about the design change. But it appears to be fine to assume that keys are globally unique.

**Comment by pedronis** [comment]
@mvo5 thanks, yes, this was agreed with Gustavo in the device registration branch but indeed it has some consequences in the assertion service itself, I will talk about them with Matt Goodall when he is back (should be tomorrow)

File: asserts/database.go [hunk 1]
```diff
@@ -149,7 +148,7 @@ func OpenDatabase(cfg *DatabaseConfig) (*Database, error) {
 		bs = nullBackstore{}
 	}
 	if keypairMgr == nil {
-		panic("database cannot be used without setting a keypair manager")
+		keypairMgr = NewMemoryKeypairManager()
```
Comment (niemeyer) [inline asserts/database.go]: :+1:

**Comment by niemeyer** [comment]
LGTM

**Comment by emgee** [comment]
+1

**Pull request merged**